HttpConnection.php

```diff
@@ -311,6 +311,46 @@ protected function isWindows() {
   }
 
   /**
+   * Returns the file size (in bytes) as a string (for 32-bit PHP).
+   *
+   * 32-bit PHP can't handle file sizes larger than 2147483647 bytes (2.15GB),
+   * since that's the PHP_INT_MAX. In order to compensate, this function
+   * uses exec() to retrieve the file size from the operating system and return
+   * it as a string. Note that converting the value back into an integer
+   * will reintroduce the same max-integer problems.
+   *
+   * Based on the function sizeExec() from https://github.com/jkuchar/BigFileTools
+   *
+   * @return string | bool (FALSE upon failure or when exec() is disabled)
+   */
+  protected function filesize_php32bit($file) {
+    $disabled_functions = explode(',', ini_get('disable_functions'));
+
+    // Ensure PHP is capable of executing an external program.
+    if ((function_exists("exec")) || (!in_array('exec', $disabled_functions))) {
+
+      $file = drupal_realpath($file);
```

Comment:
@jonathangreen i agree. Tuque should be kinda independent, but on the other side having very tight tied libraries is a tradition that is still kept alive in our community. Maybe we could break this? I Wonder what would be the best choice here. Adding maybe a config to drupal file system location? drupal realpath is a wrapper to stream wrappers, but only local system, right? I would say open a jira ticket and lets discuss that. 👍